--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -3,6 +3,11 @@ LIST(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 INCLUDE(GSettings)
 SET(ENV{LC_ALL} "C")
 
+SET(CMAKE_CXX_COMPILER clang-mp-18)
+SET(CMAKE_C_COMPILER clang-mp-18)
+SET(CMAKE_CXX_STANDARD 11)
+SET(CMAKE_INSTALL_PREFIX /opt/local)
+cmake_policy(SET CMP0100 NEW)
 
 # Project properties
 PROJECT(gimagereader)
@@ -47,7 +52,7 @@ ADD_DEFINITIONS(-DPACKAGE_REVISION=\"${PACKAGE_REVISION}\")
 ADD_DEFINITIONS(-DGETTEXT_PACKAGE=\"${CMAKE_PROJECT_NAME}\")
 ADD_DEFINITIONS(-DENABLE_VERSIONCHECK=${ENABLE_VERSIONCHECK})
 ADD_DEFINITIONS(-DMANUAL_DIR=\"${MANUAL_DIR}\")
-ADD_DEFINITIONS(-std=c++11)
+#ADD_DEFINITIONS(-std=c++11)
 
 
 # Dependencies
@@ -227,7 +232,10 @@ TARGET_LINK_LIBRARIES(gimagereader
     ${TESSERACT_LDFLAGS}
     ${gimagereader_LIBS}
     ${SANE_LDFLAGS}
+    -L/opt/local/lib/libomp
     -ldl
+    -lintl
+    -lstdc++
     -lgomp
 )
 SET_TARGET_PROPERTIES(gimagereader PROPERTIES COMPILE_FLAGS "-fopenmp")
