--- gtk/src/Recognizer.cc.orig
+++ gtk/src/Recognizer.cc
@@ -29,8 +29,8 @@
 #include <csignal>
 #include <cstring>
 #include <tesseract/baseapi.h>
-#include <tesseract/strngs.h>
-#include <tesseract/genericvector.h>
+//#include <tesseract/strngs.h>
+//#include <tesseract/genericvector.h>
 #include <tesseract/ocrclass.h>
 #include <unistd.h>
 #include <setjmp.h>
@@ -42,7 +42,7 @@
 
 
 struct Recognizer::ProgressMonitor : public MainWindow::ProgressMonitor {
-	ETEXT_DESC desc;
+	tesseract::ETEXT_DESC desc;
 	bool canceled = false;
 	int donePages = 0;
 	int nPages;
@@ -127,11 +127,11 @@ Recognizer::Recognizer() {
 std::vector<Glib::ustring> Recognizer::getAvailableLanguages() const {
 	tesseract::TessBaseAPI tess;
 	initTesseract(tess);
-	GenericVector<STRING> availLanguages;
+	std::vector<std::string> availLanguages;
 	tess.GetAvailableLanguagesAsVector(&availLanguages);
 	std::vector<Glib::ustring> result;
 	for(int i = 0; i < availLanguages.size(); ++i) {
-		result.push_back(availLanguages[i].string());
+		result.push_back(availLanguages[i]);
 	}
 	return result;
 }
@@ -432,12 +432,13 @@ void Recognizer::recognize(const std::vector<int> &pages, bool autodetectLayout)
 		}
 		OutputEditor::ReadSessionData* readSessionData = MAIN->getOutputEditor()->initRead(tess);
 		ProgressMonitor monitor(pages.size());
+                tesseract::ETEXT_DESC mdesc;
 		MAIN->showProgress(&monitor);
 		Utils::busyTask([&] {
 			int npages = pages.size();
 			int idx = 0;
 			for(int page : pages) {
-				monitor.desc.progress = 0;
+				mdesc.progress = 0;
 				++idx;
 				Glib::signal_idle().connect_once([=] { MAIN->pushState(MainWindow::State::Busy, Glib::ustring::compose(_("Recognizing page %1 (%2 of %3)"), page, idx, npages)); });
 
@@ -454,7 +455,7 @@ void Recognizer::recognize(const std::vector<int> &pages, bool autodetectLayout)
 				for(const Cairo::RefPtr<Cairo::ImageSurface>& image : MAIN->getDisplayer()->getOCRAreas()) {
 					tess.SetImage(image->get_data(), image->get_width(), image->get_height(), 4, image->get_stride());
 					tess.SetSourceResolution(MAIN->getDisplayer()->getCurrentResolution());
-					tess.Recognize(&monitor.desc);
+					tess.Recognize(&mdesc);
 					if(!monitor.canceled) {
 						MAIN->getOutputEditor()->read(tess, readSessionData);
 					}
@@ -483,6 +484,7 @@ bool Recognizer::recognizeImage(const Cairo::RefPtr<Cairo::ImageSurface> &img, O
 	}
 	tess.SetImage(img->get_data(), img->get_width(), img->get_height(), 4, 4 * img->get_width());
 	ProgressMonitor monitor(1);
+        tesseract::ETEXT_DESC mdesc;
 	MAIN->showProgress(&monitor);
 	if(dest == OutputDestination::Buffer) {
 		OutputEditor::ReadSessionData* readSessionData = MAIN->getOutputEditor()->initRead(tess);
@@ -490,7 +492,7 @@ bool Recognizer::recognizeImage(const Cairo::RefPtr<Cairo::ImageSurface> &img, O
 		readSessionData->angle = MAIN->getDisplayer()->getCurrentAngle();
 		readSessionData->resolution = MAIN->getDisplayer()->getCurrentResolution();
 		Utils::busyTask([&] {
-			tess.Recognize(&monitor.desc);
+			tess.Recognize(&mdesc);
 			if(!monitor.canceled) {
 				MAIN->getOutputEditor()->read(tess, readSessionData);
 			}
@@ -500,7 +502,7 @@ bool Recognizer::recognizeImage(const Cairo::RefPtr<Cairo::ImageSurface> &img, O
 	} else if(dest == OutputDestination::Clipboard) {
 		Glib::ustring output;
 		if(Utils::busyTask([&] {
-		tess.Recognize(&monitor.desc);
+		tess.Recognize(&mdesc);
 			if(!monitor.canceled) {
 				char* text = tess.GetUTF8Text();
 				output = text;
