--- qt/src/Recognizer.cc.orig
+++ qt/src/Recognizer.cc
@@ -29,8 +29,8 @@
 #include <sstream>
 #include <tesseract/baseapi.h>
 #include <tesseract/ocrclass.h>
-#include <tesseract/strngs.h>
-#include <tesseract/genericvector.h>
+//#include <tesseract/strngs.h>
+//#include <tesseract/genericvector.h>
 #include <QMouseEvent>
 #include <unistd.h>
 #include <setjmp.h>
@@ -49,7 +49,7 @@
 #include "ui_PageRangeDialog.h"
 
 struct Recognizer::ProgressMonitor : public MainWindow::ProgressMonitor {
-	ETEXT_DESC desc;
+	tesseract::ETEXT_DESC desc;
 	bool canceled = false;
 	int donePages = 0;
 	int nPages;
@@ -107,11 +107,11 @@ Recognizer::Recognizer(const UI_MainWindow& _ui) :
 QStringList Recognizer::getAvailableLanguages() const {
 	tesseract::TessBaseAPI tess;
 	initTesseract(tess);
-	GenericVector<STRING> availLanguages;
+	std::vector<std::string> availLanguages;
 	tess.GetAvailableLanguagesAsVector(&availLanguages);
 	QStringList result;
 	for(int i = 0; i < availLanguages.size(); ++i) {
-		result.append(availLanguages[i].string());
+		result.append(availLanguages[i].c_str());
 	}
 	return result;
 }
